pkg/deployer/lib/extension/state.go

```diff
@@ -0,0 +1,30 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package extension
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// ReconciliationState represents the current state of the reconciliation.
+// All hooks will receive the pointer to the same state object and are allowed to modify it.
+type ReconciliationState struct {
+	// ReconcileResult can be used to end the reconcile loop.
+	ReconcileResult reconcile.Result
+
+	// If set to false, reconciliation will be aborted with returning ReconcileResult after the current execution.
+	ContinueReconcile bool
+
+	// GenericState can be used to transfer any state between hooks which doesn't fit in the other fields.
+	GenericState map[string]interface{}
```

Comment:
why do we need a generic state?
this state seems hard to debug and we do not really know who writes the data and who reads it